src/main/java/org/sonarlint/intellij/core/ProjectBindingManager.java

```diff
@@ -49,10 +49,12 @@ public ProjectBindingManager(Project project, SonarLintEngineManager engineManag
   }
 
   /**
-   * Will get a Facade with the appropriate engine (standalone or connected) based on the current project and module configurations.
-   * In case of a problem, it handles the displaying of errors (Logging, user notifications, ..) and throws an IllegalStateException.
+   * Returns a Facade with the appropriate engine (standalone or connected) based on the current project and module configurations.
+   * In case of a problem, it handles the displaying of errors (Logging, user notifications, ..) and throws an InvalidBindingException.
+   *
+   * @throws InvalidBindingException If current project binding is invalid
    */
-  public synchronized SonarLintFacade getFacadeForAnalysis() {
+  public synchronized SonarLintFacade getFacade() throws InvalidBindingException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.sonarlint.intellij.exception.InvalidBindingException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)